bokehjs/src/lib/models/widgets/tables/data_table.ts

```diff
@@ -247,19 +265,51 @@ export class DataTableView extends WidgetView {
       reorderable = false
     }
 
+    let frozenRow = -1
+    let frozenBottom = false
+    const frozenColumn = this.model.frozen_columns == null ? -1 : this.model.frozen_columns-1
+    if (this.model.frozen_rows != null) {
+      frozenBottom = this.model.frozen_rows < 0 ? true : false
+      frozenRow = Math.abs(this.model.frozen_rows)
+    }
+
+    let autosize: string
+    if (this.model.fit_columns === true)
+      autosize = AutosizeModes.force_fit
+    else if (this.model.fit_columns === false)
+      autosize = AutosizeModes.off
+    else
+      autosize = (AutosizeModes as any)[this.model.autosize_mode]
+
     const options = {
       enableCellNavigation: this.model.selectable !== false,
       enableColumnReorder: reorderable,
-      forceFitColumns: this.model.fit_columns, // TODO: update to autosizeColsMode
+      autosizeColsMode: autosize,
       multiColumnSort: this.model.sortable,
       editable: this.model.editable,
-      autoEdit: false,
+      autoEdit: this.model.auto_edit,
+      autoHeight: false,
       rowHeight: this.model.row_height,
+      frozenColumn: frozenColumn,
+      frozenRow: frozenRow,
+      frozenBottom: frozenBottom
     }
 
+    if (this.model.autosize_mode === "fit_columns")
+      this.el.style.width = this.model.width + "px";
+
     this.data = new TableDataProvider(this.model.source, this.model.view)
     this.grid = new SlickGrid(this.el, this.data, columns, options)
 
+    this.grid.autosizeColumns()
+
+    if (this.model.autosize_mode === "fit_viewport") {
+      let width = 0;
+      for (const column of columns)
+        width += (column as any).width
+      this.model.width = Math.ceil(width)
+    }
```

Comment:
This is also weird, basically the idea with fit_viewport is that the viewport is adjusted to match the computed width of the columns. This approach was easiest but maybe I should just override `box_sizing` in this case and modify the width there?